src/forecast_solar/models.py

```diff
@@ -142,12 +144,10 @@ def peak_production_time(self, specific_date: date) -> datetime:
             (watt for date, watt in self.watts.items() if date.date() == specific_date),
             default=None,
         )
-        for (
-            timestamp,
-            watt,
-        ) in self.watts.items():
+        for timestamp, watt in self.watts.items():
             if watt == value:
                 return timestamp
+        return None
```

Comment:
This will give some mypy issues in the future. I could set the return type `datetime | None`, but that also causes issues elsewhere who call the function 🤔

Reply:
The `return None` is not necessary because we always expect a value, it is mainly Ruff who wants something to come here.

Reply:
Then I'd raise eg RuntimeError.

Reply:
I don't see how that would solve it or how that would be in the code.

Reply:
Instead of returning Nome here, raise an exception. Then the return value won't be optional anymore.